src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/MySqlEngine.java

```diff
@@ -753,6 +753,11 @@ protected void dropReferringFks(DbEntity entity) throws DatabaseEngineException
 
     @Override
     protected boolean checkConnection(final Connection conn) {
+        return checkConnection(conn, this.properties.getCheckConnectionTimeout());
+    }
+
+    @Override
+    protected boolean pingConnection(final Connection conn) {
```

Comment:
WDYT of making pingConnection a default method in the interface with the most common implementation and just change in the implementations where it is different (I think it's only on db2). 

Due to your findings, we don't need a pingConnection on MySQL and Oracle. Did you confirm that we don't suffer from the same problem in the other implementations. Nice job finding this, though

Reply:
Oracle is also different, but yeah you got a good point there! On it.